### PR TITLE
chore: roadmap 2025

### LINKED ISSUE #1485: Pepr 2025 Rough Roadmap
- Deno2
- System for viewing / referencing trends in performance
- Transaction Pepr Store
- OTEL
```[tasklist]
### Tasks
```

### PR DESCRIPTION
**Pull request by cmwylie19**
## Description

Start the 2025 roadmap, collaborate with team.

## Related Issue

Fixes #1485 

Relates to #

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Other (security config, docs update, etc)

## Checklist before merging
- [x] Unit, [Journey](https://github.com/defenseunicorns/pepr/tree/main/journey), [E2E Tests](https://github.com/defenseunicorns/pepr-excellent-examples), [docs](https://github.com/defenseunicorns/pepr/tree/main/docs), [adr](https://github.com/defenseunicorns/pepr/tree/main/adr) added or updated as needed
- [x] [Contributor Guide Steps](https://docs.pepr.dev/main/contribute/#submitting-a-pull-request) followed

### CODE REVIEW

File: docs/090_roadmap/README.md [hunk 1]
```diff
@@ -66,3 +66,35 @@ _2024 Roadmap_
     - Load test Pepr/KFC to identify bottlenecks and areas of improvement.
     - Ensure that Pepr/KFC can handle a large number of resources and events over a sustained period of time (nightly).
 
+_2025 Roadmap_  
+## Phase 1: Code Quality - Experimentation 
+
+- **Q1**:
+  - **Eliminate Circular Dependencies, Complexity, and strongly type return statements**:
+    - Turn on eslint settings to keep codebase clean, maintainable and standardized.
```
Comment (cmwylie19) [inline docs/090_roadmap/README.md]:
```suggestion
  - **Turn on eslint enforcement and configure settings and see no warnings**:
    - Eliminate circular dependencies, complexity, return statements, etc.
```

File: docs/090_roadmap/README.md [hunk 2]
```diff
@@ -66,3 +66,35 @@ _2024 Roadmap_
...
+  - **Metric and Performance Baselining**:
+    - Establish a baseline for performance and resource utilization metrics. Use this data to make informed decisions about the direction of the project in terms of Deno2
+  - **Experimentation with Deno2**:
+    - Experiment with Deno2 through Dash Days and see if it can be used in the project. Look into the performance improvements and new features that Deno2 brings to the table. Attempt to back the PeprStore through a Valkey Cache.
+  - **OTEL Preparation**:
+    - Come up with a plan to implement Open Telemetry. Specifically distributed tracing, metrics, logs and events. Use this data to make debugging easier from a UDS Core prespective. There will be documentation work on how to use an OTEL collector with a Pepr Module.
+
```
Comment (cmwylie19) [inline docs/090_roadmap/README.md:81]: nightly release cadence

Comment (cmwylie19) [reply docs/090_roadmap/README.md:81]: Semantic release?

File: docs/090_roadmap/README.md [hunk 3]
```diff
@@ -66,3 +66,35 @@ _2024 Roadmap_
...
+## Phase 2: Durable Storage for Metrics and Performance Tests / Transactional Pepr Store
+
+- **Q2**:
+  - **Professional Dashboard displaying metrics and performance tests originating from CI**:
+    - Begin engaging with potential contributors and users through social media, Kubernetes/Cloud Native Computing Foundation (CNCF) meetups, and other channels. Monitor and participate in our Slack channel.
```
Comment (cmwylie19) [inline docs/090_roadmap/README.md]:
```suggestion
```
delete me

something like GitHub Pages - d3 charting

File: docs/090_roadmap/README.md [hunk 4]
```diff
@@ -66,3 +66,35 @@ _2024 Roadmap_
...
+  - **Transactional Pepr Store Planning and Implementation**:
```
Comment (cmwylie19) [inline docs/090_roadmap/README.md]: Determine if it makes sense. Implement in Q3

File: docs/090_roadmap/README.md [hunk 5]
```diff
@@ -66,3 +66,35 @@ _2024 Roadmap_
...
+   - Sus out details involved with having a transactional Pepr Store. What are the implications of this? What are the benefits? What are the drawbacks? What are the use-cases? What are the technologies that can be used to implement this?
+  - **Deno2**:
+    - Depending on Q1, make a decision of whether or not to go with Deno2. If so, begin the migration process.
```
Comment (cmwylie19) [inline docs/090_roadmap/README.md]: Deno discovery to q2 -> Implementation to q3 if it happens

File: docs/090_roadmap/README.md [hunk 6]
```diff
@@ -66,3 +66,35 @@ _2024 Roadmap_
     - Load test Pepr/KFC to identify bottlenecks and areas of improvement.
     - Ensure that Pepr/KFC can handle a large number of resources and events over a sustained period of time (nightly).
 
+_2025 Roadmap_  
+## Phase 1: Code Quality - Experimentation 
+
+- **Q1**:
+  - **Eliminate Circular Dependencies, Complexity, and strongly type return statements**:
```
Comment (cmwylie19) [inline docs/090_roadmap/README.md]: add business value to bullets

File: docs/090_roadmap/README.md [hunk 7]
```diff
@@ -66,3 +66,41 @@ _2024 Roadmap_
...
+  - **Nightly Release**:
+    - Establish a nightly release process. This will help us to catch bugs early and ensure that the project is always in a releasable state.
+
+## Phase 2: Durable Storage for Metrics and Performance Tests / Transactional Pepr Store
+
+- **Q2**:
+  - **Semantic Versioning**:
+    - Establish a clear versioning scheme and adhere to it. This will help users to understand the impact of changes and make it easier to upgrade.
+  - **Professional Dashboard displaying metrics and performance tests originating from CI**:
+    - Begin engaging with potential contributors and users through social media, Kubernetes/Cloud Native Computing Foundation (CNCF) meetups, and other channels. Monitor and participate in our Slack channel.
```
Comment (btlghrants) [inline docs/090_roadmap/README.md]:
```suggestion
```
Copy pasta!

File: docs/090_roadmap/README.md [hunk 8]
```diff
@@ -66,3 +66,41 @@ _2024 Roadmap_
...
+  - **Determine if a Transactional PeprStore makes sense**:
+   - Sus out details involved with having a transactional Pepr Store. What are the implications of this? What are the benefits? What are the drawbacks? What are the use-cases? What are the technologies that can be used to implement this?
```
Comment (btlghrants) [inline docs/090_roadmap/README.md]:
```suggestion
    - Sus out details involved with having a transactional Pepr Store. What are the implications of this? What are the benefits? What are the drawbacks? What are the use-cases? What are the technologies that can be used to implement this?
```

File: docs/090_roadmap/README.md [hunk 9]
```diff
@@ -66,3 +66,41 @@ _2024 Roadmap_
...
+  - **Experimentation with Deno2**:
+    - Experiment with Deno2 through Dash Days and see if it can be used in the project. Look into the performance improvements and new features that Deno2 brings to the table. 
+
+
+## Phase 3: TBD
+
+- **Q3**:
+  - **Deno2 Implementation**:
+    - Move forward with migrating the project to Deno2. This phase will focus on adapting the codebase, conducting extensive testing, and creating comprehensive documentation to ensure a seamless transition.
```
Comment (btlghrants) [inline docs/090_roadmap/README.md]:
```suggestion
    - If determined to be advisable, move forward with migrating the project to Deno2 (starting with the kubernetes-fluent-client..?). This phase will focus on adapting the codebase, conducting extensive testing, and creating comprehensive documentation to ensure a seamless transition.
```

File: docs/090_roadmap/README.md [hunk 10]
```diff
@@ -66,3 +66,40 @@ _2024 Roadmap_
...
+  - **OTEL Preparation**:
+    - Come up with a plan to implement Open Telemetry. Specifically distributed tracing, metrics, logs and events. Use this data to make debugging easier from a UDS Core prespective. There will be documentation work on how to use an OTEL collector with a Pepr Module.
+  - **Nightly Release**:
+    - Establish a nightly release process. This will help us to catch bugs early and ensure that the project is always in a releasable state.
+
+## Phase 2: Durable Storage for Metrics and Performance Tests / Transactional Pepr Store
+
+- **Q2**:
+  - **Semantic Versioning**:
+    - Establish a clear versioning scheme and adhere to it. This will help users to understand the impact of changes and make it easier to upgrade.
```
Comment (samayer12) [inline docs/090_roadmap/README.md]:
```suggestion

```

We already use semver and it's a pre-req for Nightly releases that increment the patch version for us. I think we can delete this bullet point.

**Review by samayer12 — DISMISSED** [review]

**Review by samayer12 — APPROVED** [review]
Had a sync discussion with the team, good to go.

**Pull request merged**